src/cargo_communication/request_actor.rs

```diff
@@ -0,0 +1,226 @@
+                    return;
+                }
+                Event::CargoFinish => {
+                    self.finish_request();
+                    return;
+                }
+                Event::CargoEvent(message) => {
+                    // handle information and create notification based on that
+                    match message {
+                        CargoMessage::CargoStdout(msg) => self.handle_cargo_information(msg),
+                        CargoMessage::CargoStderr(msg) => {
+                            self.log_message(MessageType::Error, msg, None)
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    fn start_compile_task(&mut self) {
+        self.state.compile_state.start_time = get_current_time().unwrap();
+        self.report_task_start(
+            self.state.compile_state.task_id.clone(),
+            None,
+            // TODO change to actual BuildTargetIdentifier
+            Some(TaskDataWithKind::CompileTask(CompileTaskData::default())),
+        );
+    }
+
+    fn finish_request(&mut self) {
+        let res = self.cargo_handle.take().unwrap().join();
+        let status_code = get_request_status_code(&res);
+
+        self.finish_execution_task(&status_code);
+        self.report_task_finish(
+            self.state.root_task_id.clone(),
+            status_code.clone(),
+            None,
+            None,
+        );
+        self.send(Message::Response(self.create_response(res, &status_code)));
+    }
+
+    fn finish_execution_task(&self, status_code: &StatusCode) {
+        match &self.state.task_state {
+            TaskState::Compile => (),
+            TaskState::Run(run_state) => self.report_task_finish(
+                run_state.task_id.clone(),
+                status_code.clone(),
+                Some("Finished target execution".to_string()),
+                None,
+            ),
+            TaskState::Test(test_state) => self.report_task_finish(
+                test_state.task_id.clone(),
+                status_code.clone(),
+                Some("Finished target testing".to_string()),
+                None,
+            ),
+        }
+    }
+
+    fn create_response(
+        &self,
+        result: io::Result<ExitStatus>,
+        status_code: &StatusCode,
+    ) -> Response {
+        Response {
+            id: self.req_id.clone(),
+            result: result.ok().map(|_| {
+                to_value(R::Result::create_result(
+                    self.params.origin_id(),
+                    status_code.clone(),
+                ))
+                .unwrap()
+            }),
+            // TODO create error for response
+            error: None,
+        }
+    }
+
+    fn cancel(&mut self) {
+        if let Some(cargo_handle) = self.cargo_handle.take() {
+            self.cancel_process(cargo_handle);
+            self.cancel_task_and_request();
+        } else {
+            todo!("trzeba wyslac ze Task sie nie powiodl")
```

Comment:
to przeoczenie że jest niezaimplementowane?

Reply:
Nie, error handling jeszcze kompletnie nie jest zaimplementowany. Na razie mamy wersję zakładającą, że wszystko idzie bez problemu